bugsnag-plugin-android-ndk/src/test/cpp/test_serializer.c

```diff
@@ -124,15 +124,35 @@ bugsnag_event * loadSessionTestCase(jint num) {
 }
 
 bugsnag_event * loadBreadcrumbsTestCase(jint num) {
-    bugsnag_event *data = malloc(sizeof(bugsnag_event));
-    data->crumb_count = 1;
-    data->crumb_first_index = 0;
-    data->breadcrumbs[0].type = BSG_CRUMB_USER;
-    strcpy(data->breadcrumbs[0].name, "Jane");
-    strcpy(data->breadcrumbs[0].timestamp, "2018-10-08T12:07:09Z");
-    strcpy(data->breadcrumbs[0].metadata->key, "foo");
-    strcpy(data->breadcrumbs[0].metadata->value, "bar");
-    return data;
+    bugsnag_event *event = malloc(sizeof(bugsnag_event));
```

Comment:
Can we include a test where the first index isn't 1 or amend this one to start in the middle? It feels like there is a degree of complexity that we might have missed.

Reply:
I have altered this test case to add 4 breadcrumbs which start at an index of `BUGSNAG_CRUMBS_MAX - 2`, which should cover this complexity.